src/network.zig

```diff
@@ -145,24 +139,84 @@ pub fn init() void {
 	}
 }
 
-pub const Address = struct {
-	ip: u32,
+pub const AddressParseMode = enum {parse, resolve, parseOrResolve};
```

Comment:
parseOrResolve is redundant, since as far as I know resolve already handles normal IP parsing as well.
Also I'd suggest to call it `parseIp` and `resolveHostname` to be more explicit.